pkg/verifier/gcp/entry_point.go

```diff
@@ -0,0 +1,114 @@
+package gcpverifier
+
+import (
+	"encoding/base64"
+	"fmt"
+	"math/rand"
+	"strconv"
+	"time"
+
+	"github.com/openshift/osd-network-verifier/pkg/output"
+	"github.com/openshift/osd-network-verifier/pkg/verifier"
+)
+
+const cloudImageIDDefault = "cos-97-lts"
+
+// validateEgress performs validation process for egress
+// Basic workflow is:
+// - prepare for ComputeService instance creation
+// - create instance and wait till it gets ready, wait for gcpUserData script execution
+// - find unreachable endpoints & parse output, then terminate instance
+// - return `g.output` which stores the execution results
+func (g *GcpVerifier) ValidateEgress(vei verifier.ValidateEgressInput) *output.Output {
+	g.Logger.Debug(vei.Ctx, "Using configured timeout of %s for each egress request", vei.Timeout.String())
+	//default gcp machine e2
+	if vei.InstanceType == "" {
+		vei.InstanceType = "e2-standard-2"
```

Comment:
I'm getting the following when I execute it against gcp, was there a parameter that I missed to set?
(I've tried taking a built from _main_, it fails again using the same parameters but with a different error :facepalm: )

```shell
GOOGLE_APPLICATION_CREDENTIALS=~/gcp/mycredentials.json GCP_PROJECT_ID="boranccs1" GCP_VPC_NAME=boran-test-gcp-6cqhb-network ./osd-network-verifier egress --gcp --subnet-id boran-test-gcp-6cqhb-master-subnet
Using Project ID
Summary:
printing out errors faced during the execution:
 -  network verifier error: instance type e2-standard-2 is invalid: unable to gather list of supported instance types from ComputeService: Get "https://compute.googleapis.com/compute/v1/projects//zones/us-east1-b/machineTypes?alt=json&prettyPrint=false": oauth2: cannot fetch token: 400 Bad Request
Response: {"error":"invalid_scope","error_description":"Invalid OAuth scope or ID token audience provided."}
Failure!
```


```shell
# against latest main branch 
printing out errors faced during the execution:
 -  network verifier error: Unable to create labels: googleapi: Error 400: Invalid value for field 'labels': ''. Label key 'Name' violates format constraints. The key must start with a lowercase character, can only contain lowercase letters, numeric characters, underscores and dashes. The key can be at most 63 characters long. International characters are allowed., invalid <nil>
```

Reply:
yeah, probably there's a regression
I picked a random commit from the past and executed the same command, the labeling passed
```
git co 29ec86bc8ecbcf716e999b5b075fab2049a5b006

commit 29ec86bc8ecbcf716e999b5b075fab2049a5b006
Merge: f5ab1e1 9a7fcb7
Author: OpenShift Merge Robot <openshift-merge-robot@users.noreply.github.com>
Date:   Fri Sep 16 10:11:31 2022 +0200

    Merge pull request #132 from mjlshen/OSD-13005

    Break out `ec2:CreateTags` to simplify error handling logic
 
    
```

```
Using Project ID boranccs1
Using region: us-east1
Created instance with ID: verifier-4226
Applying labels
Successfully applied labels
ComputeService Instance: verifier-4226 RUNNING
Gathering and parsing console log output...
Terminating ComputeService instance with id verifier-4226
Summary:
printing out exceptions preventing the verifier from running the specific test:
 -  egressURL error: internet connectivity problem: please ensure there's internet access in given vpc subnets
Failure!
```